src/main/java/org/twilmes/sql/gremlin/processor/SingleQueryExecutor.java

```diff
@@ -381,5 +348,149 @@ SqlGremlinQueryResult convertResult(final List<Object> rows, final RelNode input
                 columnTypes.add((col == null || col.getType() == null) ? "string" : col.getType());
             }
         }
+
+        public void assertIsEmpty() {
+            synchronized (assertEmptyLock) {
+                if (currThread != null && blockingQueueRows.size() == 0) {
```

Comment:
Do we need to check if `size() == 0` here? I think it works either way, just wondering if it adds anything.

Reply:
Yea we have to make sure all results have been taken out of the queue, else the interrupt is called right when the other thread calls assertIsEmpty(), and if the read is slower then we don't get all results.